config/v1/types_console.go

```diff
@@ -52,5 +52,6 @@ type ConsoleAuthentication struct {
 	// provides the user the option to perform single logout (SLO) through the identity
 	// provider to destroy their single sign-on session.
 	// +optional
+	// +kubebuilder:validation:Pattern=^$|^((https):\/\/?)[^\s()<>]+(?:\([\w\d]+\)|([^[:punct:]\s]|\/?))$
```

Comment:
I am very curious how we test this :)

Reply:
There is a way to test that it works it terms of updating the validation in the CRD:
1) Locally, run make update-codegen 
2) Look at the CRD and the pattern should be there now in the validation section auto-magically
3) Then test as we normally do to see if the pattern works

Reply:
@sttts @damemi is this valid from kubebuilder perspective ?

Reply:
I think that's fine, I don't know how else we could test it. I did forget to mention to @zherman0 the other day that there is a manual patching ability so that you can make specific crd changes without needing to add the tag to the api (sorry), but this seems universally applicable so it looks good here

you could also add a pattern test to the toy types in kubebuilder and just use their tests but that would probably have to be a more generic tag